sheep/gateway.c

```diff
@@ -71,6 +71,11 @@ static void *init_erasure_buffer(struct request *req, int buf_len)
 	uint64_t tail = round_down(off + len, SD_EC_DATA_STRIPE_SIZE);
 	int ret;
 
+	buf = valloc(buf_len);
+	if(unlikely(!buf))
+		return NULL;
+	memset(buf, 0, buf_len);
```

Comment:
Sorry but I cannot say yes or no.
Why I do zero-clear is because xvalloc() do so:
https://github.com/sheepdog/sheepdog/blob/master/lib/util.c#L100

To remove this line safely,
we have to trace functions which touch buf (and equivalent variables)
and confirm that they do NOT assume buf to be zero-cleared.
I think it takes some times to trace.